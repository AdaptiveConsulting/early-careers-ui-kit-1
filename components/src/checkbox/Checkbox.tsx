import React from "react"
import "./checkbox.css"
interface CheckboxProps {
  size?: "small" | "medium"
  checkboxType: string
  isDisabled: boolean
  text: string
  isDarkTheme: boolean
}
export const Checkbox = ({
  size,
  checkboxType,
  isDisabled,
  isDarkTheme,
}: CheckboxProps) => {
  let accentColor = "accent-primary-light"
  const base = size === "small" ? "w-3 h-3" : "w-4 h-4"

  switch (checkboxType) {
    case "Primary":
      accentColor = isDarkTheme ? "accent-primary-dark" : "accent-primary-light"
      break
    case "Secondary":
      accentColor = isDarkTheme
        ? "accent-secondary-dark"
        : "accent-secondary-light"
      break
    case "Error":
      accentColor = isDarkTheme ? "accent-error-dark" : "accent-error-light"
      break
    case "Success":
      accentColor = isDarkTheme ? "accent-success-dark" : "accent-success-light"
      break
    case "Warning":
      accentColor = isDarkTheme ? "accent-warning-dark" : "accent-warning-light"
      break
  }

  const disabled = isDisabled ? true : false
  return (
    <input
      type="checkbox"
      disabled={disabled}
      className={`${accentColor} ${base}`}
    ></input>
  )
}
