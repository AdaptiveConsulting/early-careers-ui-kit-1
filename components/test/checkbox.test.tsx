import React from "react"
import { assert, expect, test } from "vitest"
import renderer from "react-test-renderer"
import { Checkbox } from "../src/checkbox/Checkbox"

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test("Primary checkbox component on light theme.", () => {
  const component = renderer.create(<Checkbox />)
  const tree = toJson(component)
  //expect(tree).toMatchSnapshot()
  expect(tree.props.className).toBe("accent-primary-light w-4 h-4")
})

test("Secondary checkbox component on light theme.", () => {
  const component = renderer.create(<Checkbox checkboxType="Secondary" />)
  const tree = toJson(component)
  //expect(tree).toMatchSnapshot()
  expect(tree.props.className).toBe("accent-secondary-light w-4 h-4")
})

test("Error checkbox component on light theme.", () => {
  const component = renderer.create(<Checkbox checkboxType="Error" />)
  const tree = toJson(component)
  //expect(tree).toMatchSnapshot()
  expect(tree.props.className).toBe("accent-error-light w-4 h-4")
})
test("Success checkbox component on light theme.", () => {
  const component = renderer.create(<Checkbox checkboxType="Success" />)
  const tree = toJson(component)
  //expect(tree).toMatchSnapshot()
  expect(tree.props.className).toBe("accent-success-light w-4 h-4")
})
test("Warning checkbox component on light theme.", () => {
  const component = renderer.create(<Checkbox checkboxType="Warning" />)
  const tree = toJson(component)
  expect(tree.props.className).toBe("accent-warning-light w-4 h-4")
})
test("Primary checkbox component on dark theme.", () => {
  const component = renderer.create(
    <Checkbox checkboxType="Primary" isDarkTheme={true} />,
  )
  const tree = toJson(component)
  expect(tree.props.className).toBe("accent-primary-dark w-4 h-4")
})
test("Primary, disabled checkbox component on dark theme with small size.", () => {
  const component = renderer.create(
    <Checkbox
      checkboxType="Primary"
      isDarkTheme={true}
      isDisabled={true}
      size="small"
    />,
  )
  const tree = toJson(component)
  console.log(tree)
  expect(tree.props.className).toBe("accent-primary-dark w-3 h-3")
  expect(tree.props.disabled).toBe(true)
})
