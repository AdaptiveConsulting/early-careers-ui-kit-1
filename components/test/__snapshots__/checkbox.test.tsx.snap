// Vitest Snapshot v1

exports[`Error checkbox component on light theme. 1`] = `
<input
  className="accent-error-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Link changes the class when hovered 1`] = `
<input
  className="accent-primary-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Link changes the class when hovered 2`] = `
<input
  className="accent-secondary-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Primary checkbox component on dark theme. 1`] = `
<input
  className="accent-error-dark w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Primary checkbox component on light theme. 1`] = `
<input
  className="accent-primary-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Secondary checkbox component on light theme. 1`] = `
<input
  className="accent-secondary-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Success checkbox component on light theme. 1`] = `
<input
  className="accent-success-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;

exports[`Warning checkbox component on light theme. 1`] = `
<input
  className="accent-error-light w-4 h-4"
  disabled={false}
  type="checkbox"
/>
`;
